File: network-configuration-server/src/main/java/en/ubb/networkconfiguration/dto/LayerDto.java
package en.ubb.networkconfiguration.dto;

import en.ubb.networkconfiguration.domain.enums.LayerType;
import lombok.*;
import org.nd4j.linalg.activations.Activation;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class LayerDto {

    private long id;

    private int nInputs;

    private int nNodes;

    private int nOutputs;

    private LayerType type;

    private Activation activation;

    private List<NodeDto> nodes;
}

File: client-app/src/app/shared/models/network/runtime/LayerDto.ts
import {NodeDto} from './NodeDto';
import {LayerType} from '../shared/LayerType';
import {Activation} from '../shared/Activation';

export interface LayerDto {
  id: number;
  nInputs: number;
  nNodes: number;
  nOutputs: number;
  type: LayerType;
  activation: Activation;
  nodes: NodeDto[];
}
